### ISSUE TITLE
no-mutable-property-of-props-interface: quoted property keys are never reported

### ISSUE THREAD

**Issue by ren-yamanashi** [ren-yamanashi/eslint-cdk-plugin#542]
### Describe the bug

`no-mutable-property-of-props-interface` skips every property whose key is not a plain identifier (`property.key.type !== Identifier` → `continue`). Quoted keys are therefore never checked for `readonly`, although the readonly requirement applies to them just as much.

**Repro:**

```ts
export interface QuotedKeyProps {
  "bucket-name": string; // not reported
  control: string;       // control: reported
}
```

### Expected Behavior

Both properties should be reported (and the existing fix — prefixing `readonly ` — works for quoted keys unchanged).

### Current Behavior

```
repro/no-mutable-property-of-props-interface/quoted-key.ts  3:3  error  Property 'control' of Construct Props should be readonly  awscdk/no-mutable-property-of-props-interface
```

`"bucket-name"` is not reported. Oxlint behaves identically.

### Plugin

eslint-plugin-awscdk, oxlint-plugin-awscdk

### Plugin Version

eslint-plugin-awscdk 4.4.0, oxlint-plugin-awscdk 1.0.0

### Language Version

TypeScript 6.0.3 (eslint-plugin-awscdk) / TypeScript 7 preview via oxlint-tsgolint (oxlint-plugin-awscdk)

### Additional Context

- Verified on `main` (35b889a), rule-isolated configs.
- Source: `packages/*/src/rules/no-mutable-property-of-props-interface.ts` — the key-type guard only needs to affect the report *data* (property name extraction), not whether the property is validated. The same `Identifier`-only guard exists in `require-jsdoc` / `require-props-default-doc`, where quoted keys silently skip documentation checks as well.
